cocos/core/director.ts

```diff
@@ -877,14 +877,14 @@ export class Director extends EventTarget {
      * @en Run main loop of director
      * @zh 运行主循环
      */
-    public mainLoop (time: number) {
+    public mainLoop (time?: number) {
```

Comment:
`mainLoop` is the function that we want to deprecated in the future, because it's not using a regular tick parameter. So I think we should move actual implementation to new standard function `tick` and use `tick` in mainLoop

    public mainLoop (time?: number) {
        if (!this._invalid) {
            // calculate "global" dt
            this.calculateDeltaTime(time);
            const dt = this._deltaTime;

            this.tick(dt);
        }
    }